scripts/Inelastic/IndirectReductionCommon.py

```diff
@@ -1009,6 +1058,54 @@ def get_multi_frame_rebin(workspace_name, rebin_string):
 # -------------------------------------------------------------------------------
 
 
+def _get_x_range_when_bins_vary(workspace: MatrixWorkspace, grouping_workspace: GroupingWorkspace) -> Tuple[float]:
+    """
+    Finds the minimum and maximum X values for a workspace with varying bins. It will only search for the min
+    and max x values in spectra which are included in the grouping provided by the GroupingWorkspace.
+    @param workspace The workspace to search for the min and max X.
+    @param grouping_workspace The workspace containing detector grouping information.
+    @return A minimum and maximum X value.
+    """
+    excluded_ids = _excluded_detector_ids(grouping_workspace)
+
+    min_value, max_value = float("inf"), float("-inf")
+    spec_info = workspace.spectrumInfo()
+    for i in range(workspace.getNumberHistograms()):
+        if spec_info.isMasked(i) or not spec_info.hasDetectors(i):
+            continue
+        detector_ids = workspace.getSpectrum(i).getDetectorIDs()
+        if detector_ids[0] in excluded_ids:
+            continue
+        x = workspace.extractX()[i]
+        min_value = x.min() if x.min() < min_value else min_value
+        max_value = x.max() if x.max() > max_value else max_value
+
+    assert min_value > 0, "The minimum x value in a dSpacing workspace should be a positive number."
+    assert max_value == float("-inf") or max_value > 0, "The maximum x value in a dSpacing workspace should be a positive number."
+
+    return min_value, max_value
+
+
+def rebin_logarithmic(workspace: MatrixWorkspace, calibration_file: str) -> MatrixWorkspace:
+    """
+    Performs logarithmic rebinning on the provided workspace. The rebinning parameters are calculated
+    based on the spectra which are included in the grouping provided in the calibration file.
+    @param workspace The workspace to be logarithmically rebinned.
+    @param calibration_file The *.cal file to use to find which detectors are excluded from the grouping.
+    @return A rebinned workspace.
+    """
+    grouping_workspace = create_grouping_workspace(workspace, calibration_file)
+
+    min_value, max_value = _get_x_range_when_bins_vary(workspace, grouping_workspace)
+
+    if min_value == float("inf") or max_value == float("-inf"):
+        raise RuntimeError("No selected Detectors found in .cal file for input range.")
+
+    step = expm1((log(max_value) - log(min_value)) / workspace.blocksize())
+
+    return Rebin(InputWorkspace=workspace, Params=[min_value, step, max_value], BinningMode="Logarithmic", StoreInADS=False)
```

Comment:
I replaced the use of the `DiffractionFocussing` algorithm because it was using the calibration file to do grouping. In this algorithm, it was also doing logarithmic rebinning (I don't know why)
https://github.com/mantidproject/mantid/blob/36cb74141910af5a5146bc5011c654f6aca53668/Framework/Algorithms/src/DiffractionFocussing.cpp#L214

So to get the same results as previously, I needed to do logarithmic rebinning before applying the requested grouping. I can add something to the documentation to mention it does logarithmic rebinning

Reply:
Probably is something related to resolution. I guess it won't hurt to specify it somewhere.